src/Storages/MergeTree/MergeTreeDataWriter.cpp

```diff
@@ -588,7 +599,13 @@ MergeTreeDataWriter::TemporaryPart MergeTreeDataWriter::writeTempPartImpl(
 
     for (const auto & projection : metadata_snapshot->getProjections())
     {
-        auto projection_block = projection.calculate(block, context);
+        Block projection_block;
+        {
+            ProfileEventTimeIncrement<Microseconds> watch(ProfileEvents::MergeTreeDataWriterProjectionsCalculationMicroseconds);
```

Comment:
Maybe also add it to https://github.com/ClickHouse/ClickHouse/blob/7e77c52c75940a4ce9fc77801d4d934d81028712/src/Storages/MergeTree/MutateTask.cpp#L1245?

Reply:
not sure though how it could be used. afaiu this metric won't be associated with any query, so you only could query it through `system.metric_log`, like sum in a time interval?

Reply:
We have `ProfileEvents` field in `system.part_log`. It should go there for `MutatePart` entry.